### PR TITLE
python plugin fixes for better tracing debugging

### PR DESCRIPTION
**Pull request by ktsaou**
@paulfantom it was not starting in my systems.

I did a few changes, mainly to logging, parsing of update_every (was not working) and detecting the `python.d` directory (was not working either).

I am missing a very important feature: I need to verify it has parsed everything properly and the main data collection loop works as expected. This can be verified if we add `debug` statements that log all key actions and timings. Unfortunately, I am not that good in python. Actually, I am a little bit confused. Could you please implement these?

I am not merging this, until you review it. Once this is merged, you can continue your work.

### CODE REVIEW

**Comment by ktsaou** [comment]
Also, it would be nice if we could use `debug()`, `error()`, `info()` and `fatal()` in python.d modules.

File: plugins.d/python.d.plugin [hunk 1]
```diff
@@ -479,18 +504,15 @@ def parse_cmdline(directory, *commands):
...
+                update_every = UPDATE_EVERY
 
     debug("started from", commands[0], "with options:", *commands[1:])
-    if len(mods) == 0 and DEBUG_FLAG is False:
-        interval = None
 
-    return {'interval': interval,
-            'modules': mods}
+    UPDATE_EVERY = update_every
+    return {'modules': mods}
```
Comment (paulfantom) [inline plugins.d/python.d.plugin]: Now it is not needed to wrap `mods` into dict. Just change lines in `run()` from:
```
out = parse_cmdline(MODULES_DIR, *sys.argv)
modules = out['modules']
```
to:
```
modules = parse_cmdline(MODULES_DIR, *sys.argv)
```

File: plugins.d/python.d.plugin [hunk 2]
```diff
@@ -505,42 +527,40 @@ def run():
     disabled = []
     configfile = CONFIG_DIR + "python.d.conf"
 
-    interval = INTERVAL
+    update_every = UPDATE_EVERY
     conf = read_config(configfile)
     if conf is not None:
         try:
+            # FIXME: is this right? exit the whole plugin?
```
Comment (paulfantom) [inline plugins.d/python.d.plugin]: This is intended.
If you set `enabled: no` in `python.d.conf` it disables whole python driver

File: python.d/example.chart.py [hunk 3]
```diff
@@ -1,16 +1,18 @@
 # Description: example netdata python.d plugin
 # Author: Pawel Krupa (paulfantom)
 
+import os
+import sys
```
Comment (paulfantom) [inline python.d/example.chart.py]: `sys` is not needed here.

File: plugins.d/python.d.plugin (hunk 1 above)
Comment (ktsaou) [inline plugins.d/python.d.plugin]: It breaks and I don't know how to fix it. Do this yourself after being merged?

File: plugins.d/python.d.plugin [hunk 4]
```diff
@@ -7,21 +7,79 @@ import os
...
+
 # setup environment
 # https://github.com/firehol/netdata/wiki/External-Plugins#environment-variables
-MODULES_DIR = os.getenv('NETDATA_PLUGINS_DIR',
-                        os.path.abspath(__file__).strip("python.d.plugin.py").replace("plugins.d", ""))
+MODULES_DIR = os.path.abspath(os.getenv('NETDATA_PLUGINS_DIR',
+                        os.path.dirname(__file__)) + "/../python.d") + "/"
 CONFIG_DIR = os.getenv('NETDATA_CONFIG_DIR', "/etc/netdata/")
-INTERVAL = os.getenv('NETDATA_UPDATE_EVERY', None)
+UPDATE_EVERY = os.getenv('NETDATA_UPDATE_EVERY', 1)
```
Comment (paulfantom) [inline plugins.d/python.d.plugin]: This way if you run `python.d.plugin debug 1` every python job will run every second.
Look at the end of `__init__()` in `PythonCharts`

File: python.d/example.chart.py (hunk 3 above)
Comment (ktsaou) [inline python.d/example.chart.py]: done

File: plugins.d/python.d.plugin (hunk 1 above)
Comment (paulfantom) [inline plugins.d/python.d.plugin]: Ok, merge it. I will fix it.

File: plugins.d/python.d.plugin (hunk 4 above)
Comment (ktsaou) [inline plugins.d/python.d.plugin]: Ok, let me state how it should work (like the rest of netdata):

1. The environment variable `NETDATA_UPDATE_EVERY` is the default update_every for every plugin (it should default to 1, if unset)
2. A numeric command line argument overwrites this global default
3. The option `update_every` in `python.d.conf` overwrites this global default
4. The option `update_every` in `python.d/MODULE.chart.py` sets the default for all jobs of a module (if unset the global default will be set as module default)
5. The option `update_every` in a job in `python.d/MODULE.chart.py` sets the update frequency of the specific job. If unset, the module default will be used as the job update_every.

I am not sure how to fix it. Do you want me to cancel this PR and use it as a base to fix it?

**Comment by ktsaou** [comment]
merge or cancel?

**Comment by paulfantom** [comment]
Merge and I will fix it in the next hours.

**Comment by ktsaou** [comment]
thanks!

File: plugins.d/python.d.plugin (hunk 4 above)
Comment (paulfantom) [inline plugins.d/python.d.plugin]: It was like this in the beginning. But you wanted to override this with 'debug' command line parameter, like this: `python.d.plugin debug mysql 1` that's why I run it with default value 'None' and later use many checks.

**Pull request merged**